PIMS/Controllers/ProductsController.cs

```diff
@@ -0,0 +1,152 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Services.Helper;
+using ServicesContract.Model;
+using ServicesContracts;
+using ServicesContracts.Model;
+using System;
+using System.Threading.Tasks;
+
+namespace Assignment_1.Controllers
+{
+    [Route("/products")]
+    public class ProductsController : Controller
+    {
+        private readonly IProducts _products;
+        private readonly ExportExcel _exportExcel;
+
+        public ProductsController(IProducts products, ExportExcel exportExcel)
+        {
+            _products = products;
+            _exportExcel = exportExcel;
+        }
+
+        [HttpGet("/")]
+        public async Task<string> Home()
+        {
+            return "Hello World";
+        }
+
+
+        /// <summary>
+        /// Endpoint to add a new product.
+        /// </summary>
+        /// <param name="product">The model containing product details.</param>
+        /// <returns>Returns the result of adding the product.</returns>
+       
+        [HttpPost("add")]
+        public async Task<IActionResult> AddProducts([FromBody] ProductsModel product)
+        {
+            try
+            {
+                var result = await _products.AddProducts(product);
+                if (result)
+                {
+                    return Ok("Product added");
+                }
+                return BadRequest("Product not added");
+            }
+            catch (Exception ex)
+            {
+                return BadRequest(ex.Message);
+            }
+        }
+
+
+        /// <summary>
+        /// Endpoint to retrieve all products.
+        /// </summary>
+        /// <returns>Returns the list of all products.</returns>
+       
+        [HttpGet("getall")]
+        public async Task<IActionResult> GetProducts()
+        {
+            try
+            {
+                var result = await _products.GetProducts();
+                return Ok(result);
+            }
+            catch (Exception ex)
+            {
+                return BadRequest(ex.Message);
```

Comment:
Added Logging to Catch Blocks for Better Debugging and Error Tracking
Why: Logging exceptions helps in diagnosing issues by providing detailed information about errors that occurred. This is crucial for understanding the cause and context of errors, especially in production environments.

Reply:
Avoided Exposing Raw Exception Messages to the Client, Instead Returning a Generic Error Message While Logging the Actual Exception
Why: Exposing raw exception messages can potentially leak sensitive information about the server or application, which could be exploited by malicious users. Instead, log the detailed exception and return a generic error message to the client.

Reply:
Resolved